rero/sonar
# security: add permissions for accessing resources

* Creates a factory for managing permissions.
* Adds serializer and loader for deposit resources.
* Adds a deny access permission.
* Adds specific permissions decisions for all resources.
* Adds queries factories for filtering records in lists.
* Adds permissions results when serializing a record.
* Changes path in deposit validation email.
* Guesses user's organisation when creating a user record.
* Guesses user's organisation when creating a document record.
* Removes redirection to documents records when accessing administration.
* Adds permissions checks for adding resources when retrieving logged user by API.
* Removes organisation from user and document schema when logged user is not superuser.
* Improves fixtures in tests.
* Closes #146.
* Closes #217.

## How to test
For each roles (superuser, admin, moderator, publisher, user):
1. Login as user with the current role.
2. Check the access of the different resources. It must correspond to permissions matrix in https://docs.google.com/spreadsheets/d/1GyFbkm3VDCrffbShaFsoMJnkGQYtw65sWbfW25J2Mxk/edit#gid=0.
